packages/auth/src/core/auth/auth_impl.ts

```diff
@@ -567,43 +564,35 @@ export class AuthImpl implements AuthInternal, _FirebaseService {
       return;
     }
 
-    this.idTokenSubscription.next(this.currentUser);
+    this.idTokenEmitter.emit(this.currentUser);
 
     const currentUid = this.currentUser?.uid ?? null;
     if (this.lastNotifiedUid !== currentUid) {
       this.lastNotifiedUid = currentUid;
-      this.authStateSubscription.next(this.currentUser);
+      this.authStateEmitter.emit(this.currentUser);
     }
   }
 
-  private registerStateListener(
-    subscription: Subscription<User>,
-    nextOrObserver: NextOrObserver<User>,
-    error?: ErrorFn,
-    completed?: CompleteFn
-  ): Unsubscribe {
+  private createUserStateStream(
```

Comment:
Thanks for creating this PR and for the thorough analysis!

I wonder if a smaller scoped change like below would work (i have not tried this, just thinking out aloud). I also posted this on the issue:

 private registerStateListener(
    subscription: Subscription<User>,
    nextOrObserver: NextOrObserver<User>,
    error?: ErrorFn,
    completed?: CompleteFn
  ): Unsubscribe {
    if (this._deleted) {
      return () => {};
    }

+ let isUnsubscribed = false;
    const cb =
      typeof nextOrObserver === 'function'
        ? nextOrObserver
        : nextOrObserver.next.bind(nextOrObserver);

    const promise = this._isInitialized
      ? Promise.resolve()
      : this._initializationPromise;
    _assert(promise, this, AuthErrorCode.INTERNAL_ERROR);
    // The callback needs to be called asynchronously per the spec.
    // eslint-disable-next-line @typescript-eslint/no-floating-promises
    + promise.then(() => {if (isUnsubscribed) { return; }; cb(this.currentUser)});

    if (typeof nextOrObserver === 'function') {
      + Unsubscribe unsubscribe = subscription.addObserver(nextOrObserver, error, completed);
      + return {isUnsubscribed = true; unsubscribe();}
    } else {
      + Unsubscribe unsubscribe = subscription.addObserver(nextOrObserver);
      + return {isUnsubscribed = true; unsubscribe();}
    }
  }
```

Since the Observer logic is also used outside of Auth (in storage and firestore), i wonder if we can do a smaller fix. Thanks!